packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -196,29 +199,153 @@ class CupertinoAlertDialog extends StatelessWidget {
       children.add(const Padding(padding: const EdgeInsets.only(top: 8.0)));
     }
 
+    return new ClipRRect(
```

Comment:
Do we need this? It'll end up getting clipped by the outermost cliprrect anyway right? (Just because there's a cost to these clips)

Reply:
we shouldn't have any clips. Just draw the right shape.

Reply:
Might be tricky with BackdropFilter

Reply:
the top corners of the content, and all corners of the backdrop filter have been given corner radii by way of a container decoration instead of ClipRRect. There is still one remaining ClipRRect on the bottom because the bottom actions are arbitrary developer-provided widgets.

Reply:
We can define a protocol for telling those widgets which corner they're in so that they can do the right thing, then we can make the widgets we provide implement it without a clip.

We want to get rid of all clips. They are prohibitively expensive.